osu.Game/Screens/Play/HUD/UnstableRateCounter.cs

```diff
@@ -0,0 +1,183 @@
+            scoreProcessor.JudgementReverted += onJudgementReverted;
+        }
+
+        private bool isValid;
+
+        private void setValid(bool valid)
+        {
+            if (isValid == valid) return;
+
+            DrawableCount.FadeTo(valid ? 1 : alpha_when_invalid, 1000, Easing.OutQuint);
+            isValid = valid;
+        }
+
+        private void onJudgementAdded(JudgementResult judgement)
+        {
+            if (!(judgement.HitObject.HitWindows is HitWindows.EmptyHitWindows) && judgement.IsHit)
+            {
+                hitOffsets.Add(judgement.TimeOffset);
+            }
+
+            updateUr();
+        }
+
+        // If a judgement was reverted successfully, remove the item from the hitOffsets list.
+        private void onJudgementReverted(JudgementResult judgement)
+        {
+            //Score Processor Conditions to revert
+            if (judgement.FailedAtJudgement || !judgement.Type.IsScorable())
+                return;
+            //UR Conditions to Revert
+            if (judgement.HitObject.HitWindows is HitWindows.EmptyHitWindows || !judgement.IsHit)
+                return;
+
+            hitOffsets.RemoveAt(hitOffsets.Count - 1);
+            updateUr();
+        }
+
+        private void updateUr()
+        {
+            // At Count = 0, we get NaN, While we are allowing count = 1, it will be 0 since average = offset.
+            if (hitOffsets.Count > 0)
+            {
+                double mean = hitOffsets.Average();
+                double squares = hitOffsets.Select(offset => Math.Pow(offset - mean, 2)).Sum();
+                Current.Value = Math.Sqrt(squares / hitOffsets.Count) * 10;
+                setValid(true);
+            }
+            else
+            {
+                Current.Value = 0;
+                setValid(false);
+            }
+        }
+
+        protected override LocalisableString FormatCount(double count)
+        {
+            return count.ToString("0.00 UR");
+        }
+
+        protected override IHasText CreateText() => new TextComponent
+        {
+            Alpha = alpha_when_invalid,
+        };
+
+        protected override void Dispose(bool isDisposing)
+        {
+            base.Dispose(isDisposing);
+
+            if (scoreProcessor == null) return;
+
+            scoreProcessor.NewJudgement -= onJudgementAdded;
+            scoreProcessor.JudgementReverted -= onJudgementReverted;
+        }
+
+        private class TextComponent : CompositeDrawable, IHasText
+        {
+            public LocalisableString Text
+            {
+                get => intPart.Text;
+                set
+                {
+                    //Not too sure about this, is there a better way to go about doing this?
+                    splitValue = value.ToString().Split('.');
+                    intPart.Text = splitValue[0];
+                    decimalPart.Text = splitValue[1];
+                }
```

Comment:
I could live with the setter but the getter is pretty bad. would store the value received via setter in a field and return that.

not a huge deal because nobody ideally should be reading it but feels cleaner.